src/editors/colorpicker.js

```diff
@@ -0,0 +1,61 @@
+/*
+
+Edtended handling of  oolor type fields.
+
+Has optional support for using https://github.com/Sphinxxxx/vanilla-picker.
+
+*/
+import { StringEditor } from './string'
+import { $extend } from '../utilities'
+
+export var ColorEditor = StringEditor.extend({
+
+  build: function () {
+    this._super()
+    if (!this.input) return
+    if (window.Picker) { // do when vanilla-picker loaded
+      this.input.type = 'text'
+      const self = this
+      const defaultValue = this.schema.default || '#000000'
+      const pickerOptions = $extend({
```

Comment:
You should use the new method of accessing the plugin options, like this:

````javascript
      options = this.expandCallbacks('colorpicker', $extend({}, {
        height: 300 // You can place any default plugin options here
      }, this.defaults.options.colorpicker || {}, this.options.colorpicker || {}))
````
Using this method you can define global options for the plugins as` JSONEditor.defaults.options.colorpicker` and/or in schema options as `options.colorpicker` (This overrides the global options)
And it also allows you to use/define JavaScript callback functions using `JSONEditor.defaults.callbacks.colorpicker`